lisa/tools/cargo.py

```diff
@@ -9,14 +11,33 @@
 from lisa.tools.echo import Echo
 from lisa.tools.gcc import Gcc
 from lisa.tools.ln import Ln
-from lisa.util import UnsupportedDistroException
+from lisa.tools.rm import Rm
+from lisa.util import LisaException, UnsupportedDistroException
 from lisa.util.process import ExecutableResult
 
 
 class Cargo(Tool):
+    # cargo 1.67.1 (8ecd4f20a 2023-01-10)
+    _version_pattern = re.compile(
+        r"cargo (?P<major>\d+).(?P<minor>(\d+)).(?P<patch>(\d+)) ",
+        re.M,
+    )
+
+    # stable-x86_64-unknown-linux-gnu (default)
+    _rust_toolchain_pattern = re.compile(
+        r"\w+-\w+-\w+-\w+-\w+",
+        re.M,
+    )
+    toolchain: str = ""
+    __command: str = "cargo"
+
     @property
     def command(self) -> str:
-        return "cargo"
+        return self.__command
```

Comment:
have you tried to use self.command?

Reply:
Hi Lili, yes i tried with setter of command property. Now i have used _initialize() to set self._command as we discussed. Please check and let me know.